src/main/java/com/cs3365/rpgdapi/weapons/WeaponsController.java

```diff
@@ -47,4 +48,15 @@ public WeaponsController(WeaponsService weaponsService) {
         return ResponseEntity.created(location)
                              .body("");
     }
+    
+    @RequestMapping(value = "/files/{id}")
+    public ResponseEntity<WeaponEntity> get(@PathVariable("id") UUID identifier)
+    {
+        try {
+            return ResponseEntity.ok(weaponsService.findWeapon(identifier));
```

Comment:
This may be a decent shortcut. I usually try to set the response as a variable first, but if this works, great.